contracts/src/protocol/OmniAVSStorage.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity =0.8.12;
+
+import { IAVSDirectory } from "eigenlayer-contracts/src/contracts/interfaces/IAVSDirectory.sol";
+
+import { IDelegationManager } from "../interfaces/IDelegationManager.sol";
+import { IOmniPortal } from "../interfaces/IOmniPortal.sol";
+import { IOmniAVS } from "../interfaces/IOmniAVS.sol";
+
+abstract contract OmniAVSStorage {
+    /// @notice Constant used as a divisor in calculating weights
+    uint256 internal constant WEIGHTING_DIVISOR = 1e18;
+
+    /// @notice EigenLayer core DelegationManager
+    IDelegationManager internal immutable _delegationManager;
+
+    /// @notice EigenLayer core AVSDirectory
+    IAVSDirectory internal immutable _avsDirectory;
+
+    /// @notice Strategy parameters for restaking
+    IOmniAVS.StrategyParams[] internal _strategyParams;
+
+    /// @notice List of currently register operators, used to sync EigenCore
+    address[] internal _operators;
+
+    /// @notice Set of operators that are allowed to register
+    mapping(address => bool) internal _allowlist;
+
+    /// @notice Maximum number of operators that can be registered
+    uint32 public maxOperatorCount;
+
+    /// @notice Omni chain id, used to make xcalls to the Omni chain
+    uint64 public omniChainId;
+
+    /// @notice OmniPortal.xcall gas limit per each validator in syncWithOmni
+    uint64 public xcallGasLimitPerValidator = 10_000;
+
+    /// @notice OmniPortal.xcall base gas limit in syncWithOmnj
+    uint64 public xcallBaseGasLimit = 75_000;
+
+    /// @notice Minimum stake required for an operator to register, not including delegations
+    uint96 public minOperatorStake;
+
+    /// @notice Omni portal contract, used to make xcalls to the Omni chain
+    IOmniPortal public omni;
+
+    constructor(IDelegationManager delegationManager_, IAVSDirectory avsDirectory_) {
```

Comment:
this is logic so should still be in the other contract if it can be. This should just be storage variables. Then it's clear that it is an append only list of storage vars and we can add to the end of it, but can't change anything in the middle

Reply:
If we want immutables and constants in the storage contract we need this here.

So yeah guess question is the same yours above - do we want to list immutables and constants here, or in parent?

Like you said, they're not storage so yeah no real need for them to be in storage contract. Other than just organization. 

I do like having all of the "storage /  constants / immutables" in one place visually. But I won't fight for it. 

This is pattern in eigen contracts   so I think either way we're fine.

Reply:
Cool. You make the call

Reply:
I'm gonna go with immutables and constants in parent. So the storage can be only storage, and it's clear that anything there must be append only.